src/Microsoft.DotNet.Interactive.PackageManagement/KernelSupportsNugetExtensions.cs

```diff
@@ -192,59 +193,76 @@ internal static KernelCommandInvocation DoNugetRestore()
                         return;
                     }
 
-                    var install = new InstallPackagesMessage(
-                            kernel.RestoreSources.OrderBy(s => s).ToList(),
-                            kernel.ResolvedPackageReferences.Select(s => $"{s.PackageName}, {s.PackageVersion}").OrderBy(s => s).ToList(),
+                    var requestedPackages =
+                        kernel.RequestedPackageReferences
+                              .Except(kernel.ResolvedPackageReferences, PackageReferenceComparer.Instance)
+                              .Select(s => s.PackageName).OrderBy(s => s).ToList();
 
-                            Enumerable.Empty<string>().ToList(),
-                            kernel.RequestedPackageReferences
-                                  .Except(kernel.ResolvedPackageReferences, PackageReferenceComparer.Instance)
-                                  .Select(s => s.PackageName).OrderBy(s => s).ToList(), 0);
-
-                    CreateOrUpdateDisplayValue(context, installPackagesPropertyName, install);
-
-                    var currentPackageReferences = kernel.ResolvedPackageReferences?.ToHashSet() ?? new HashSet<ResolvedPackageReference>();
+                    var requestedSources =
+                        kernel.RestoreSources.OrderBy(s => s).ToList();
 
-                    var restorePackagesTask = kernel.RestoreAsync();
+                    var installMessage =
+                        new InstallPackagesMessage(
+                            requestedSources,
+                            requestedPackages,
+                            Enumerable.Empty<string>().ToList(),
+                            0);
 
-                    var delay = 500;
-                    while (await Task.WhenAny(Task.Delay(delay), restorePackagesTask) != restorePackagesTask)
+                    using (var operation = Logger.Log.OnEnterAndExit())
                     {
-                        if (context.CancellationToken.IsCancellationRequested)
+                        CreateOrUpdateDisplayValue(context, installPackagesPropertyName, installMessage);
+
+                        var currentPackageReferences = kernel.ResolvedPackageReferences?.ToHashSet() ?? new HashSet<ResolvedPackageReference>();
+                        var restorePackagesTask = kernel.RestoreAsync();
```

Comment:
It looks like cancellation was never supported here. It would be great to fix this but I don't want to hold up the PR for it.

Reply:
I'll add an issue

Reply:
https://github.com/dotnet/interactive/issues/1515